src/campaign.js

```diff
@@ -144,6 +144,32 @@ const { Util } = require("./util.js");
           faultString = faultString.trim();
       }
 
+      // https://github.com/adobe/acc-js-sdk/issues/108
+      // 401 error code is hidden in error message and incorrectly reported as HTTP 500 error causing the refresh token
+      // callback not to be called
+      // Extract specific error code if possible
+      if (errorCode == "SOP-330007" && errorMessage && errorMessage.indexOf("XSV-350114") != -1 && errorMessage.indexOf(" 401") != -1) {
```

Comment:
Use a const for SOP-330007 as IMS_TOKEN_EXPIRY_CODE and then we can add it to comparison.
Create a separate function to check if it is a token expiry errorMessage. The function will take the errorMessage, check for null and other validations. It'll return true/false based on that.

Reply:
`// Define constants for error codes
const IMS_TOKEN_EXPIRY_CODE = "SOP-330007";
const TOKEN_EXPIRED_IDENTIFIER = "XSV-350114";
const UNAUTHORIZED_CODE = " 401";

function isTokenExpiryError(errorMessage) {
    if (!errorMessage) {
        return false;  // Return false if errorMessage is null or undefined
    }

    return errorMessage.includes(TOKEN_EXPIRED_IDENTIFIER) && errorMessage.includes(UNAUTHORIZED_CODE);
}

// Main logic
if (errorCode === IMS_TOKEN_EXPIRY_CODE && isTokenExpiryError(errorMessage)) {
    // Handle token expiry logic here
}
`